tutorials/llm/llama-3/nemo2-sft-peft/nemo2-sft.ipynb

```diff
@@ -0,0 +1,632 @@
+    "        limit_val_batches=2,\n",
+    "        val_check_interval=2,\n",
+    "        num_sanity_val_steps=0,\n",
+    "    )\n",
+    "    return trainer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "### Logger\n",
+    "Configure your training steps, output directories and logging through `NeMoLogger`. In the following example, the experiment output will be saved at `./results/nemo2_sft`.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def logger() -> run.Config[nl.NeMoLogger]:\n",
+    "    ckpt = run.Config(\n",
+    "        nl.ModelCheckpoint,\n",
+    "        save_last=True,\n",
+    "        every_n_train_steps=10,\n",
+    "        monitor=\"reduced_train_loss\",\n",
+    "        save_top_k=1,\n",
+    "        save_on_train_epoch_end=True,\n",
+    "        save_optim_on_train_end=True,\n",
+    "    )\n",
+    "\n",
+    "    return run.Config(\n",
+    "        nl.NeMoLogger,\n",
+    "        name=\"nemo2_sft\",\n",
+    "        log_dir=\"./results\",\n",
+    "        use_datetime_version=False,\n",
+    "        ckpt=ckpt,\n",
+    "        wandb=None\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "### Optimizer\n",
+    "In the following example, we will be using distributed adam optimizer, and pass in optimizer configuration through `OptimizerConfig`: \n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def adam_with_cosine_annealing() -> run.Config[nl.OptimizerModule]:\n",
+    "    opt_cfg = run.Config(\n",
+    "        OptimizerConfig,\n",
+    "        optimizer=\"adam\",\n",
+    "        lr=5e-6,\n",
+    "        adam_beta2=0.98,\n",
+    "        use_distributed_optimizer=True,\n",
+    "        clip_grad=1.0,\n",
+    "        bf16=True,\n",
+    "    )\n",
+    "    return run.Config(\n",
+    "        nl.MegatronOptimizerModule,\n",
+    "        config=opt_cfg\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Base Model\n",
+    "We will perform SFT on top of Llama3-8b so we create a `LlamaModel` to pass to finetune API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def llama3_8b() -> run.Config[pl.LightningModule]:\n",
+    "    return run.Config(llm.LlamaModel, config=run.Config(llm.Llama3Config8B))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AutoResume\n",
+    "In NeMo 2.0 we can directly pass in Llama3-8b's Hugging Face ID to start SFT without manually converting it into NeMo checkpoint format like in NeMo 1.0."
```

Comment:
fix punctuation, remove possessive "s" in  Llama3-8b Hugging Face ID, add definitive article "the", and revise sentence

   "In NeMo 2.0, we can directly pass in the Llama3-8b Hugging Face ID to start SFT without manually converting it into  the NeMo checkpoint format, as required in NeMo 1.0."

Reply:
Done, thanks!